### PR TITLE
Update API in order to support the per stream state

### PR DESCRIPTION
**Pull request by benmoriceau**
## What

Update the API in order to handle the per stream state (and per stream reset).

### CODE REVIEW

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -3707,6 +3730,19 @@ components:
           format: int64
         status:
           $ref: "#/components/schemas/JobStatus"
+        streams:
+          type: array
+          items:
+            $ref: "#/components/schemas/StreamKey"
+    StreamKey:
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:3737]: reminder: replace thjs with stream descriptor

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -3987,8 +4023,100 @@ components:
           $ref: "#/components/schemas/ConnectionId"
         state:
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:4024]: switch this to legacy?

File: airbyte-api/src/main/openapi/config.yaml [hunk 3]
```diff
@@ -3987,8 +4023,100 @@ components:
...
+    StreamStateObject:
+      type: object
+      required:
+        - streamKey
+      properties:
+        streamKey:
+          $ref: "#/components/schemas/StreamKey"
+        state:
+          $ref: "#/components/schemas/ConnectionStateObject"
+    ConnectionStateType:
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:4042]: i think in the new world, we do actually need a add legacy to this enum.

File: airbyte-api/src/main/openapi/config.yaml [hunk 4]
```diff
@@ -3987,8 +4023,100 @@ components:
...
+      type: string
+      enum:
+        - global
+        - per_stream
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:4046]:
```suggestion
        - stream
```

File: airbyte-api/src/main/openapi/config.yaml [hunk 5]
```diff
@@ -3987,8 +4023,100 @@ components:
           $ref: "#/components/schemas/ConnectionId"
         state:
           $ref: "#/components/schemas/ConnectionStateObject"
+        globalState:
```
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:4026]: this is out of date.

**Review by cgardens — CHANGES_REQUESTED** [review]
started re-reviewing this. realized that there are still a few things in the spec that need ironing out. lmk if you'd like to go over any of them!

File: airbyte-api/src/main/openapi/config.yaml (hunk 2 above)
Comment (benmoriceau) [reply airbyte-api/src/main/openapi/config.yaml:4024]: I am wondering about compatibility issues. I have no idea about how this endpoint is used (apart from it not being use in the UI). I some users have a script fetching the state, it should remains the same.

File: airbyte-api/src/main/openapi/config.yaml (hunk 3 above)
Comment (benmoriceau) [reply airbyte-api/src/main/openapi/config.yaml:4050]: done

File: airbyte-api/src/main/openapi/config.yaml (hunk 5 above)
Comment (benmoriceau) [reply airbyte-api/src/main/openapi/config.yaml:4026]: done

**Comment by cgardens** [comment]
closing in favor https://github.com/airbytehq/airbyte/pull/13835 so that other people can review it.

**Pull request closed without merging**